numpy/core/memmap.py

```diff
@@ -236,11 +236,11 @@ def __new__(subtype, filename, dtype=uint8, mode='r+', offset=0,
                 raise ValueError("Size of available data is not a "
                         "multiple of the data-type size.")
             size = bytes // _dbytes
-            shape = (size,)
+            shape = (np.intp(size),)
         else:
             if not isinstance(shape, tuple):
                 shape = (shape,)
-            size = 1
+            size = np.intp(1) # Causes bytes below to be computed as largest integer type available to prevent overflow
```

Comment:
`intp` is just "the integer type that the OS uses for sizes" - it may or may not be the largest integer type (on windows, it's not)

Reply:
On amd64 this wouldn't end up as an int64?